app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -130,41 +127,52 @@ public void onLocationChanged(Location location) {
         locationFab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                if (mLocation != null && mGoogleMap != null) {
+
+                Context context = getContext();
+
+                if (mLocation != null && mGoogleMap != null && context != null) {
                     mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
                             new LatLng(mLocation.getLatitude(), mLocation.getLongitude()), 15));
 
-                    MainActivity.toast.setText("Found you!");
-                    MainActivity.toast.show();
+                    Toast.makeText(context, "Found you!", Toast.LENGTH_SHORT).show();
                 }
                 else{
 
-                    MainActivity.toast.setText("Waiting on location...");
-                    MainActivity.toast.show();
+                    if (context != null) {
+                        Toast.makeText(context, "Waiting on location...", Toast.LENGTH_SHORT).show();
+                    }
                 }
             }
         });
 
-        mView.findViewById(R.id.closeSuggestions).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                mView.findViewById(R.id.layoutSuggestions).setVisibility(View.GONE);
-            }
-        });
-
         return mView;
     }
+
+    private void showLocalizationFailed() {
+
+        Context context = getContext();
+
+        // TODO: Instead of a toast, lets show a red bar as long as there is no signal (like in the Niantic app)
+
+        // Yes!! The context actually *can* be null; e.g. when you leave the settings view while
+        // no GPS location was found. Timing matters and in the end the activity has already
+        // been destroyed; well then there is no context and guess what...
+        if (context != null) {
+            Toast.makeText(context, "No GPS signal.", Toast.LENGTH_LONG).show();
+        }
+    }
+
     private void initMap(){
         if (mLocation != null && mGoogleMap != null){
             mGoogleMap.setMyLocationEnabled(true);
             mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
                     new LatLng(mLocation.getLatitude(), mLocation.getLongitude()), 15));
-            MainActivity.toast.setText("Found you!");
-            MainActivity.toast.show();
+        } else {
+            showLocalizationFailed();
```

Comment:
according to this mGoogleMap could also be null,
can we handle both scenarios in showLocalizationFailed() ?